app/services/get_conversation_data_service.rb

```diff
@@ -0,0 +1,31 @@
+class GetConversationDataService
+  def initialize(target)
+    @target = target
+    @user = target.user
+    @conversation = target.conversation
+  end
+
+  def other_user
+    first_conversation_user = conversation_users.first
+    return first_conversation_user if first_conversation_user.id != @user.id
```

Comment:
You can add `attr_reader :user, @conversation` and you avoid using the variables.

See https://www.rubyguides.com/2018/11/attr_accessor/